python_modules/dagster/dagster/_core/definitions/declarative_automation/operators/dep_operators.py

```diff
@@ -53,6 +53,23 @@ def evaluate(self, context: AutomationContext[T_EntityKey]) -> AutomationResult[
         )
 
 
+@whitelist_for_serdes
+class AssetMatchesCondition(BuiltinAutomationCondition[T_EntityKey]):
```

Comment:
Let's call this `EntityMatchesCondition`, as you've written it in a way that is generic and can handle either assets or asset checks.

We don't want "entity" to be a user-facing term at the moment, so it's ok for the static constructor to keep the name `asset_matches`, but this is an internal class so we should give it the more accurate name.